agent/src/beerocks/monitor/monitor_thread.cpp

```diff
@@ -1484,51 +1484,63 @@ bool monitor_thread::hal_event_handler(bwl::base_wlan_hal::hal_event_ptr_t event
 
         // If event == Channel_Scan_New_Results_Ready do nothing since is_dump's default is 0
         if (event == Event::Channel_Scan_Dump_Result) {
-            auto msg = static_cast<bwl::sCHANNEL_SCAN_RESULTS_NOTIFICATION *>(data);
+            auto &msg = *(static_cast<bwl::sCHANNEL_SCAN_RESULTS_NOTIFICATION *>(data));
```

Comment:
Why? If `data` is `nullptr` it'll crash.
It's fine to work with pointers here...

Reply:
Acutally I think `data` is tested for `nullptr` above, isn't it?
Anyway looking at it again I agree that its fine to use pointers, so do what you think is best in this case.